CMakeBuild.py

```diff
@@ -60,21 +88,35 @@ def run(self):
                 "-DSUNDIALS_ROOT={}".format(os.path.abspath(self.sundials_root))
             )
 
-        if not os.path.exists(self.build_temp):
-            os.makedirs(self.build_temp)
+        build_dir = self.get_build_directory()
+        if not os.path.exists(build_dir):
```

Comment:
This if statement is equivalent to
```
os.makedirs(build_dir, exist_ok=True)
```

Reply:
Good catch

Reply:
There are a few things that need refactoring in this module, so will do that then